ch_pipeline/processing/quarterstack.py

```diff
@@ -0,0 +1,332 @@
+    data to include are made at the time the revision is *created*. At that point the
+    database is queried, good days are found, checked for existence, and they are
+    divided up into groups for each of the output stacks. These decisions are saved
+    into the `revconfig.yaml` file. When jobs to create the individual items (i.e.
+    stacks) are generated, the set of days to stack are simply pulled from this
+    revision config.
+
+    This has the important result that after the revision has been created, changes
+    to the daily data, e.g. processing new days or changing opinions about them in
+    the database, will not effect the stacks to be generated.
+    """
+
+    type_name = "quarterstack"
+    tag_pattern = r"(?P<year>\d{4})q(?P<quarter>[1-4])p(?P<partition>\d)"
+
+    # Parameters of the job processing
+    default_params = {
+        # Daily processing revisions to use (later entries in this list take precedence
+        # over earlier ones)
+        "daily_revisions": ["rev_02"],
+        "daily_root": "/project/rpp-krs/chime/chime_processed/",
+        # Frequencies to process
+        "freq": [0, 1024],
+        # The beam transfers to use (need to have the same freq range as above)
+        "product_path": "/project/rpp-krs/chime/bt_empty/chime_4cyl_allfreq/",
+        "partitions": 2,
+        # Don't generate quarter stacks with less days than this
+        "min_days": 5,
+        # RA range used to estimate the cross talk for each quarter
+        "crosstalk_ra": {
+            "q1": [165, 180],
+            "q2": [240, 255],
+            "q3": [315, 330],
+            "q4": [45, 60],
+        },
+        # Job params
+        "time": 180,  # How long in minutes?
+        "nodes": 16,  # Number of nodes to use.
+        "ompnum": 6,  # Number of OpenMP threads
+        "pernode": 8,  # Jobs per node
+    }
+
+    default_script = DEFAULT_SCRIPT
+
+    def _create_hook(self):
+        """Create the revision.
+
+        This tries to determine which days are good and bad, and partitions the
+        available good days into the individual stacks.
+        """
+
+        days = {}
+
+        core.connect()
+
+        # Go over each revision and construct the set of LSDs we should stack, and save
+        # the path to each.
+        # NOTE: later entries is `daily_revisions` will override LSDs found in earlier
+        # revisions.
+        for rev in self.default_params["daily_revisions"]:
+
+            daily_path = (
+                self.root_path
+                if self.default_params["daily_root"] is None
+                else self.default_params["daily_root"]
+            )
+            daily_rev = daily.DailyProcessing(rev, root_path=daily_path)
+
+            # Get all the bad days in this revision
+            revision = df.DataRevision.get(name=rev)
+            query = (
+                df.DataFlagOpinion.select(df.DataFlagOpinion.lsd)
```

Comment:
Should this not depend on data flags rather than opinions?
This only makes sense because at the moment there is a small number of opinions entered by people you trust to make good decisions.

Reply:
This should be how you get the lsd out of dataflags that were created by voting: https://github.com/chime-experiment/chimedb_dataflag/blob/409b18d309ecf16bdec8f306094f84ee057895dc/chimedb/dataflag/orm.py#L251
```
df.DataFlag.select(df.DataFlagVote.lsd).distinct().join(df.DataFlagVote).where(df.DataFlagVote.revision == revision, df.DataFlag.type == "vote")
```
The result is empty (there are no DataFlagVote's in the db)